### DISCUSSION TITLE
Problema con la dependencia mockito (TPI2)

### DISCUSSION THREAD

**Discussion by falvarezscarzella** [dds-jv/foro#271] [Consultas de herramientas] [answered]
Hola buenas,
Estoy hace un par de días ya intentando resolver el problema de que no me toma el Mockito como dependencia en los test, pero sin embargo esta reconocido como tal en la estructura del proyecto.

![image](https://user-images.githubusercontent.com/62475073/121184607-a4e48f00-c83b-11eb-8175-73abd2d6dd33.png)

![image](https://user-images.githubusercontent.com/62475073/121191194-20e1d580-c842-11eb-9047-24beb411b909.png)

Maven esta bien instalado, ya lo verifique.
Este problema se repite con los plugins como por ejemplo los del checkstyle que me tira warnings que deberia ignorar segun el pom como "MissingJavadocMethod, MissingJavadocPackage, MissingJavadocType".

![image](https://user-images.githubusercontent.com/62475073/121195121-b6cb2f80-c845-11eb-89cf-f11e8f5ef134.png)

En definitiva es como si no estuviera leyendo completamente el pom.
El trabajo esta hecho y los test tambien pero no tengo manera de probarlo ni de correr "mvn verify"

Saludos

**Comment by flbulgarelli** [comment, marked as answer]
Hola,

¿Cual es el enlace a tu repositorio? Me gustaría poder bajarlo y probar qué sucede al correrlo localmente. O mejor aún, mostranos que error produce en la consola al ejecutar mvn verify.

A priori me resulta extraña la combinación de ambos problemas: por un lado comentas que no anda mvn verify, lo cual hablaría de que tenés un problema con la instalación de maven, pero por otro lado mencionas que no reconoce sus dependencias en el IDE, y no son cosas que necesariamente vayan de la mano.

De hecho, honestamente no recuerdo si el plugin de checkstyle toma las exclusiones que están en el pom (en principio apostaría a que no), por lo que ese comportamiento no hablaría de un problema con maven y quizás no sean cuestiones  relacionadas.

**Reply by falvarezscarzella** [reply]
Finalmente si anda el maven verify, de hecho me tomo el mockito y corrio los test correctamente a pesar de aparecerme asi, por lo que pude entregar el trabajo sin problemas. Es un problema del IDE definitivamente. Me toma unos 50+ errores de todos los mensajes de mockito( como el mock() ) pero, sin embargo, funciona incluso corriendo los test desde el mismo IntelliJ solo que aparece asi, en rojo, como si esa biblioteca no existiera o no estuviera incluida.

**Reply by gastonprieto** [reply]
Una cosa que se me ocurre, puede ser que se este quejando porque estas haciendo el import * de mockito en vez de imports individuales de las cosas que usas?
